documentation/ADRs/020_Log_Files_and_Real-Time_Alerts.md

```diff
@@ -0,0 +1,75 @@
+# Log Files and Real-Time Alerts
+
+| ADR Info            | Details                   |
+|---------------------|---------------------------|
+| Subject             | Log files and real-time alerts |
+| ADR Number          | 020                       |
+| Status              | Proposed                  |
+| Author              | Marina                    |
+| Date                | 28.10.2024                |
+
+
+## Context
+In order to have a more effective overview of the orchestration in the pipeline, there is a need for better handling of log files and implementing real-time alerts which would allow for timely and relevant error handling for critical issues. 
+
+
+## Decision
+To implement a real-time alert and notification system which will distribute alerts through specific channels (eg. Slack, email, W&B, Prefect), targeted at particular audiences (eg. Infrastructure, MD&D, individual) depending on the type of alert being handled. 
+
+
+### Overview
+
+Alert channels: Slack, email, Prefect and Weights&Biases
+
+Logging levels: INFO, WARNING, ERROR, CRITICAL
+
+Audience: MD&D, Infrastructure, Outreach, monthly-run responsible 
+
+| Channel | Threshold        | Alert Type                         | Audience                  |
+|---------|-------------------|------------------------------------|---------------------------|
+| Slack   | ERROR, CRITICAL   | Real-Time, Short Summary + Link to Full Log | MD&D, Infrastructure      |
+| Email   | ERROR, CRITICAL   | Real-Time, Expanded Summary       | MD&D, Infrastructure      |
+| Prefect | ALL               | Real-Time, Workflow Status        | MD&D, Infrastructure, Outreach |
+| W&B     | ALL               | Performance Metrics               | MD&D, Infrastructure, Outreach |
+
+
+For alert prioritization, "critical" and "non-critical" alerts are separated, with critical alerts stemming from logging levels ERROR and CRITICAL. 
+
+
+## Consequences
+
+**Positive Effects:**
+- Real-time alerts allow for timely handling of errors which may be critical 
+- Alerts can reach their target audiences immediately which limits time and makes information flows more direct and effective 
+- Having real-time alerts is useful in generating a record of (reoccuring) issues in the pipeline 
+- Increased transparency
+
+
+**Negative Effects:**
+- Alerts must be set up in a way to be concise and informative in order to be useful, with carefully chosen levels of detail to avoid "alert fatigue"
+- The audience for each type of alert must be carefully selected in order for alerts to stay relevant 
+- Potential increase in maintanence work 
```

Comment:
Nice, I would just put a little term in front for easy reading:

**Positive Effects:**
- **Timely Issue Resolution:** Real-time alerts enable prompt handling of critical errors, reducing downtime and minimizing impact.
- **Targeted Communication:** Alerts reaching specific audiences ensure that relevant teams are immediately informed, enhancing the efficiency of response efforts.
- **Historical Insights:** Real-time alerts contribute to a log of recurring issues, facilitating trend analysis and proactive problem-solving.
- **Enhanced Transparency:** Improved logging and alerting mechanisms provide greater visibility into pipeline operations, fostering accountability and informed decision-making.

**Negative Effects:**
- **Alert Fatigue:** Without careful configuration, the volume of alerts may overwhelm recipients, leading to important alerts being overlooked.
- **Relevance of Alerts:** Incorrectly targeted alerts can result in irrelevant notifications, causing confusion and reducing trust in the alerting system.
- **Increased Maintenance:** Setting up and maintaining the alerting infrastructure requires additional resources and ongoing adjustments to thresholds and channels.